tools/c7n_azure/tests/test_storage.py

```diff
@@ -69,6 +70,79 @@ def test_firewall_rules_include(self):
         self.assertEqual(len(resources), 1)
 
     @arm_template('storage.json')
+    @cassette_name('firewall')
+    def test_firewall_rules_any(self):
+        p = self.load_policy({
+            'name': 'test-azure-storage',
+            'resource': 'azure.storage',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'ccipstorage*'},
+                {'type': 'firewall-rules',
+                 'any': ['1.2.2.128/25', '8.8.8.8', '10.10.10.10']}],
+        })
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    @arm_template('storage.json')
+    @cassette_name('firewall')
+    def test_firewall_rules_not_any(self):
+        p = self.load_policy({
+            'name': 'test-azure-storage',
+            'resource': 'azure.storage',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'glob',
+                 'value_type': 'normalize',
+                 'value': 'ccipstorage*'},
+                {'type': 'firewall-rules',
+                 'any': ['8.8.8.8', '10.10.10.10']}],
```

Comment:
Something feels weird here...

'Only' filter with '1.2.2.128/25', '3.1.1.1', '10.10.10.10' -- passes, but 'any' with '8.8.8.8', '10.10.10.10' - fails.. I'd expect them to have 10.10.10.10 being present in ip rules set of the storage